openassessment/xblock/submission_mixin.py

```diff
@@ -250,24 +279,34 @@ def save_files_descriptions(self, data, suffix=''):  # pylint: disable=unused-ar
 
         return {'success': True, 'msg': u''}
 
+    def create_team_submission(self, student_sub_data):
+        """ A student submitting for a team should generate matching submissions for every member of the team. """
+        if self.has_team():
+            team_anonymous_user_ids = self.get_team_anonymous_user_ids()
+
+            submissions = []
+
+            # submissions are identical except for the student item
+            for anonymous_user_id in team_anonymous_user_ids:
+                student_item_dict = self.get_student_item_dict(anonymous_user_id=anonymous_user_id)
+                submissions.append(self.create_submission(student_item_dict, student_sub_data))
+
+        # Since we don't get the right UUID through the above loop,
+        # null it out and fetch it from the DB during the next page load
+        self.submission_uuid = None
```

Comment:
I think you could get the correct submission_uuid if you make that change I talked about in your edx-platform PR - the one about `get_team_anonymous_user_ids()` returning a map from plain user id to their anon. id.

Reply:
So, in light of our conversation on your other PR... :)
I think you can get the anon. id of the current user at any time using `self.xmodule_runtime.anonymous_student_id`, so you could check that against the current anon. id in your loop, and if they match, set `self.submission_uuid` to the current submission's UUID.